k8s/conversion.go

```diff
@@ -0,0 +1,25 @@
+package k8s
+
+// Converter describes a type which can convert a kubernetes kind from one API version to another.
+// Typically there is one converter per-kind, but a single converter can also handle multiple kinds.
+type Converter interface {
+	// Convert converts a raw kubernetes kind into the target APIVersion.
+	// The RawKind argument will contain kind information and the raw kubernetes object,
+	// and the returned bytes are expected to be a raw kubernetes object of the same kind and targetAPIVersion
+	// APIVersion.
```

Comment:
According to the webhook documentation there is another requirement which we should probably be passing on to the `Converter` implementations (since we don't handle it ourselves):
```
The webhook is expected to set `apiVersion` of these objects to the `request.desiredAPIVersion`.
```

We could also look into simplifying the interface and making sure that we set the version ourselves, in the webhook handler code.

Reply:
Ah, I guess I wasn't explicit in my assumption that the returned object should have the correct `kind` and `apiVersion`. I'll make sure the godocs are explicit about this.